pandas/core/indexes/base.py

```diff
@@ -3953,18 +3953,37 @@ def memory_usage(self, deep: bool = False) -> int:
 
     def where(self, cond, other=None):
         """
-        Return an Index of same shape as self and whose corresponding
-        entries are from self where cond is True and otherwise are from
-        other.
+        Return a copy of self with entries replaced where cond is False.
+
+        The replacement is taken from other.
 
         Parameters
         ----------
         cond : bool array-like with the same length as self
-        other : scalar, or array-like
+            Condition to select the entries on.
+        other : scalar, or array-like, default None
+            Replacement if the condition is False.
 
         Returns
         -------
-        Index
+        pandas.Index
+            Copy of self with entries replaced from other
+            where the condition is False.
+
+        See Also
+        --------
+        DataFrame.where : Replace values in a DataFrame where the condition is False.
+
+        Examples
+        --------
+        >>> idx1 = pd.Index([42, 21, 34, 96, 72])
```

Comment:
Do you mind trying to find an example with somehow real data? Using arbitrary numbers makes this difficult to read.

For example, something like:
```python
idx = pandas.Index(['car', 'bike, 'train', 'tractor'])
idx.where(idx.isin(['car', 'train'], 'other')
```